### ISSUE TITLE
Problem when trying to use gurobi and linopy, on a .exe generated by auto-py-to-exe.

### ISSUE THREAD

**Issue by KanaBaradei** [PyPSA/linopy#382]
Dear all,

I can run my script without problems using Visual Studio Code, however when I create the .exe using autopytoexe the following error appears on the console:

![image](https://github.com/user-attachments/assets/615f0fa5-0b11-4d33-976d-a19b750d8eb3)

I was using pyomo on the past and it worked both on VSC and the .exe. I'm wondering if it is a problem with linopy not being able to call gurobi correctly in this situation?

**Comment by lkstrp** [comment]
This is not related to `linopy`, might be an issue with `pyinstaller` or is simply wrong usage.

Some things to keep in mind:
- The gurobi solver is not installed by default with `pip install linopy`. You need to install it in addition to linopy (`pip install gurobipy` or for all solvers `pip install 'linopy[solvers]'`). Please check if you do this in your exe workflow.
- Also check that `linopy` is loaded before `gurobipy` is installed. Linopy does the check for available solvers upon loading, so any installations after that will not be found.

Without more information about your script and the `pyinstaller` command, this cannot be debugged.

**Comment by KanaBaradei** [comment]
Gurobipy is installed.
I'm running the script in VSC and everything works, however when I run that same .exe on this exact same machine it throws the error shown above.
The linopy part of the script is very basic, I construct a model and then solve it.
The commands sent to pyinstaller are:
pyinstaller --noconfirm --onefile --console  "path to the file".
I already had this same thing working with pyomo and the executable worked.
Maybe there is a problem in paths or something?

**Comment by KanaBaradei** [comment]
Problem solved, for anyone that has this issue in the future:
I downloaded one of the examples of GUROBI directly from their website, ran it on Visual Studio Code, verified that it worked. Created the .exe and found that one module was missing, added it to hidden options in pyinstaller and then another one popped. After 3 iterations adding the missing modules each time it finally worked. After that I went back to my linopy script and added those modules and it finally worked.

This are the modules:
--hidden-import "gurobipy._util" --hidden-import "gurobipy._attrutil" --hidden-import "gurobipy._modelutil" 
Hope this helps anyone in the future.

@lkstrp thank you for your time. It was indeed a problem with pyinstaller and gurobi.
